src/main/java/shop/service/impl/UserServiceImpl.java

```diff
@@ -5,17 +5,21 @@
 import shop.entity.User;
 import shop.enums.UserRole;
 import shop.service.UserService;
+import shop.utils.PasswordEncoder;
 
 import java.util.Optional;
 import java.util.UUID;
 
 public class UserServiceImpl implements UserService {
+    private PasswordEncoder passwordEncoder = new PasswordEncoder();
```

Comment:
you don't need to create an instance of class. It should be utility class with static methods.
If you that, u just can use PasswordEncoder.encrypt() directly